api/src/wrangler.rs

```diff
@@ -120,11 +128,21 @@ impl<'b> ExecutionWrangler<'b> {
     }
 
     pub fn resolve_address(&self, addr: &Address) -> anyhow::Result<Option<ActorID>> {
-        self.bench.resolve_address(addr)
+        self.bench.borrow().resolve_address(addr)
     }
 
+    /// Returns a reference to the underlying blockstore
+    /// The blockstore handle here is intended to be short-lived as some executors may buffer changes leading to this
+    /// handle drifting out of sync
+    // TODO: https://github.com/anorth/fvm-workbench/issues/15
     pub fn store(&self) -> &dyn Blockstore {
-        self.bench.store()
+        // It's unfortunate that we need to call flush here everytime we need the wrangler to give out a blockstore reference
+        // However, the state_tree inside Executor wraps whatever blockstore it's given with a BufferedBlockstore
+        // Since the store held by the Wrangler was cloned prior to being wrapped in the BufferedBlockstore, it's possible
+        // there are pending changes to the underlying blockstore held in the BufferedBlockstore's cache that are not
+        // visible via our handle.
```

Comment:
I commented on the issue. Maybe we could switch to a callback pattern here (and Bench) for accessing the store?